src/components/ModalDialog/ModalDialog.tsx

```diff
@@ -98,4 +111,20 @@ export default class ModalDialog extends React.Component<ModalDialogProps> {
       </FixedGridRow>
     );
   };
+
+  private handleKeyDown = (e: KeyboardEvent) => {
+    if (e.keyCode === KeyCodes.escape) {
+      this.setState(() => ({
+        isOpen: false,
+      }));
+    }
+  };
+
+  private startKeyListener = () => {
+    document.addEventListener('keydown', this.handleKeyDown);
+  };
+
+  private stopKeyListener = () => {
+    document.removeEventListener('keydown', this.handleKeyDown);
```

Comment:
this needs to happen on componentDidUnmount too right?

Reply:
Got it.